cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/UniverseClient.scala

```
@@ -49,93 +75,233 @@ class UniverseRepositoryFetcher(adminRouter: AdminRouter)(implicit statsReceiver
+                           "marathon.json.mustache")) {
+              oldV2Package.copy(
+                  marathonMustache = Some(ByteBuffer.wrap(buffer)))
+            } else if (entryPath.getFileName() == Paths.get("config.json")) {
+              oldV2Package.copy(
+                  config = Some(
+                      parseJson(entryPath, new String(buffer)).asObject
+                        .getOrElse(
+                          throw PackageFileSchemaMismatch(
+                              "config.json", DecodingFailure("Object", List()))
+                      )
+                  )
+              )
+            } else if (entryPath.getFileName() == Paths.get("resource.json")) {
+              oldV2Package.copy(
+                  resource = Some(
+                      parseAndVerify[universe.v2.model.Resource](
+                          entryPath, new String(buffer))
+                  )
+              )
+            } else if (entryPath.getFileName() == Paths.get("command.json")) {
+              oldV2Package.copy(
+                  command = Some(
+                      parseAndVerify[universe.v2.model.Command](
+                          entryPath, new String(buffer))
+                  )
+              )
+            } else {
+              // We have an extra file for some reason. Just ignore it.
+              logger.warn(
+                  "Encountered extra file while processing Universe zip file: {}",
+                  entryPath)
+              oldV2Package
+            }
+
+          state + (((packageName, releaseVersion), newV2Package))
+      }
+
+    universe.v3.model.Repository(
+        universeRepository.toList.flatMap {
+          case ((packageName, releaseVersion), packageInfo) =>
+            packageInfo.packageDetails.map {
+              details =>
+                // This skips packages that don't contain package.json
+                val marathon = packageInfo.marathonMustache.getOrElse(
```

To be consistent with `UniversePackageCache`, we should throw an error if this file is missing.